### PR TITLE
[WebProfilerBundle] Profile the latest identical request with One-click

### PR DESCRIPTION
**Pull request by Starojitski**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | N/A
| License       | MIT
| Doc PR        | N/A

When debugging a particular request one often needs to view the most recent version of that request. That is with the same IP, URL and Method. Quite often that request is not the 'Latest' one. In that case developer has to click on button 'Last 10' and find the request in question in the list and click on it.
This pull request makes it possible to fetch profile of most recent version of the currently viewed profile with just one click. By clicking on the "refresh" button next to the "Profiled on" time stamp.

image

### CODE REVIEW

File: src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/layout.html.twig [hunk 1]
```diff
@@ -73,10 +73,15 @@
                             </dd>
 
                             <dt>Profiled on</dt>
-                            <dd>{{ profile.time|date('r') }}</dd>
+                            <dd>{{ profile.time|date('r') }}
+                                <a class="btn btn-sm" href="{{ path('_profiler', { token: 'latest', id: profile.ip, method: profile.method|upper, url: profile.url }|merge(request.query.all)) }}">
+                                    {{ include('@WebProfiler/Icon/ajax.svg') }}</a>
```
Comment (stof) [inline src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/layout.html.twig]: putting that in the `<dd>` of `Profiled on` looks weird to me.

Comment (Starojitski) [reply src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/layout.html.twig]: I agree with you, but I could not find a better place. At first i put it in the sidebar but then it required changes in CSS as well and was not clear what the button does.
Then i tried putting it next to the `Token` field in the header - like get the same request profile with the most recent token. Later i decided to put it where 'recent' is semi-implied, namely where the timestamp is shown. Even now it does not feel completely right but i decided to make a pull request and hopefully some day someone else with find a better place

**Comment by curry684** [comment]
I like the idea, but not the presentation: looks like a weird refresh button now.

Perhaps a plain left and right arrow next to the URL would be more intuitive for previous/next similar request?

**Comment by SenseException** [comment]
There seem to be no tests for that controller action yet. Shouldn't these changes, and the action itself, be covered?

**Comment by fabpot** [comment]
@Starojitski I see that you've merged master into this PR. We cannot merge a pull request containing merges. Can you rebase the pull request instead (that will get rid of the merge commit)? Thanks.

**Comment by Starojitski** [comment]
@fabpot I have no idea how to "unmerge" and how to "rebase" it. Sorry

**Comment by stof** [comment]
@Starojitski see our contributing documentation. It explains how to do it: http://symfony.com/doc/current/contributing/code/patches.html#step-3-submit-your-patch

**Comment by fabpot** [comment]
Not convinced that this is something we need to add. At first, I thought that the refresh button was to re-trigger the same HTTP request. Closing for now.

**Pull request closed without merging**